[CI] PersistentTaskInitializationFailureIT.testPersistentTasksThatFailDuringInitializationAreRemovedFromClusterState

**Build scan**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/Madrid&tests.container=org.elasticsearch.persistent.PersistentTaskInitializationFailureIT&tests.sortField=FAILED&tests.test=testPersistentTasksThatFailDuringInitializationAreRemovedFromClusterState&tests.unstableOnly=true
**Repro line**:
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.persistent.PersistentTaskInitializationFailureIT.testPersistentTasksThatFailDuringInitializationAreRemovedFromClusterState" -Dtests.seed=62B59473E8A31EC3 -Dtests.locale=sl -Dtests.timezone=America/Porto_Acre -Druntime.java=11
```
**Reproduces locally?**:
No

**Applicable branches**:
`7.x`

**Failure excerpt**:

```
org.elasticsearch.transport.TransportSerializationException: Failed to deserialize response from handler [org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler/org.elasticsearch.action.ActionListenerResponseHandler@57c55cdc/org.elasticsearch.client.transport.TransportClientNodesService$RetryListener@7937457c] |  
-- | --
  | at org.elasticsearch.transport.InboundHandler.handleResponse(InboundHandler.java:267) [main/:?] |  
  | at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:128) [main/:?] |  
  | at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:84) [main/:?] |  
  | at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:693) [main/:?] |  
  | at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:129) [main/:?] |  
  | at org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:104) [main/:?] |  
  | at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:69) [main/:?] |  
  | at org.elasticsearch.transport.nio.MockNioTransport$MockTcpReadWriteHandler.consumeReads(MockNioTransport.java:311) [framework-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at org.elasticsearch.nio.SocketChannelContext.handleReadBytes(SocketChannelContext.java:217) [elasticsearch-nio-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at org.elasticsearch.nio.BytesChannelContext.read(BytesChannelContext.java:29) [elasticsearch-nio-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at org.elasticsearch.nio.EventHandler.handleRead(EventHandler.java:128) [elasticsearch-nio-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at org.elasticsearch.transport.nio.TestEventHandler.handleRead(TestEventHandler.java:140) [framework-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at org.elasticsearch.nio.NioSelector.handleRead(NioSelector.java:409) [elasticsearch-nio-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at org.elasticsearch.nio.NioSelector.processKey(NioSelector.java:235) [elasticsearch-nio-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at org.elasticsearch.nio.NioSelector.singleLoop(NioSelector.java:163) [elasticsearch-nio-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at org.elasticsearch.nio.NioSelector.runLoop(NioSelector.java:120) [elasticsearch-nio-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT] |  
  | at java.lang.Thread.run(Thread.java:834) [?:?] |  
  | Caused by: java.lang.IllegalArgumentException: Unknown NamedWriteable category [org.elasticsearch.persistent.PersistentTaskParams] |  
  | at org.elasticsearch.common.io.stream.NamedWriteableRegistry.getReader(NamedWriteableRegistry.java:96) ~[main/:?] |  
  | at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:34) ~[main/:?] |  
  | at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:28) ~[main/:?] |  
  | at org.elasticsearch.persistent.PersistentTasksCustomMetadata$PersistentTask.<init>(PersistentTasksCustomMetadata.java:340) ~[main/:?] |  
  | at org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:656) ~[main/:?] |  
  | at org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:640) ~[main/:?] |  
  | at org.elasticsearch.persistent.PersistentTasksCustomMetadata.<init>(PersistentTasksCustomMetadata.java:525) ~[main/:?] |  
  | at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:35) ~[main/:?] |  
  | at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:28) ~[main/:?] |  
  | at org.elasticsearch.cluster.metadata.Metadata.readFrom(Metadata.java:1034) ~[main/:?] |  
  | at org.elasticsearch.cluster.ClusterState.readFrom(ClusterState.java:694) ~[main/:?] |  
  | at org.elasticsearch.action.admin.cluster.state.ClusterStateResponse.lambda$new$0(ClusterStateResponse.java:37) ~[main/:?] |  
  | at org.elasticsearch.common.io.stream.StreamInput.readOptionalWriteable(StreamInput.java:1037) ~[main/:?] |  
  | at org.elasticsearch.action.admin.cluster.state.ClusterStateResponse.<init>(ClusterStateResponse.java:37) ~[main/:?] |  
  | at org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:58) ~[main/:?] |  
  | at org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:25) ~[main/:?] |  
  | at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1264) ~[main/:?] |  
  | at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1251) ~[main/:?] |  
  | at org.elasticsearch.transport.InboundHandler.handleResponse(InboundHandler.java:264) ~[main/:?] |  
  | ... 16 more
```

--- a/server/src/internalClusterTest/java/org/elasticsearch/persistent/PersistentTaskInitializationFailureIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/persistent/PersistentTaskInitializationFailureIT.java
@@ -12,6 +12,7 @@ import org.elasticsearch.Version;
 import org.elasticsearch.action.support.PlainActionFuture;
 import org.elasticsearch.client.Client;
 import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.metadata.Metadata;
 import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.UUIDs;
 import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
@@ -41,7 +42,13 @@ public class PersistentTaskInitializationFailureIT extends ESIntegTestCase {
         return Collections.singletonList(FailingInitializationPersistentTasksPlugin.class);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/71995")
+    @Override
+    protected Collection<Class<? extends Plugin>> transportClientPlugins() {
+        // We need to pass the plugin to the transport client in order to get FailingInitializationTaskParams registered
+        // into the transport client NamedWriteableRegistry
+        return Collections.singletonList(FailingInitializationPersistentTasksPlugin.class);
+    }
+
     public void testPersistentTasksThatFailDuringInitializationAreRemovedFromClusterState() throws Exception {
         PersistentTasksService persistentTasksService = internalCluster().getInstance(PersistentTasksService.class);
         PlainActionFuture<PersistentTasksCustomMetadata.PersistentTask<FailingInitializationTaskParams>> startPersistentTaskFuture =
@@ -54,9 +61,8 @@ public class PersistentTaskInitializationFailureIT extends ESIntegTestCase {
         startPersistentTaskFuture.actionGet();
 
         assertBusy(() -> {
-            final ClusterService clusterService = internalCluster().getMasterNodeInstance(ClusterService.class);
-            final PersistentTasksCustomMetadata persistentTasks =
-                clusterService.state().metadata().custom(PersistentTasksCustomMetadata.TYPE);
+            Metadata metadata = client().admin().cluster().prepareState().execute().actionGet().getState().getMetadata();
+            final PersistentTasksCustomMetadata persistentTasks = metadata.custom(PersistentTasksCustomMetadata.TYPE);
             assertThat(persistentTasks.tasks().toString(), persistentTasks.tasks(), empty());
         });
     }

